boot/boot.go

```diff
@@ -40,67 +39,96 @@ type BootParticipant interface {
 	// ChangeRequiresReboot returns whether a reboot is required to switch
 	// to the snap.
 	ChangeRequiresReboot() bool
+	// Is this a trivial implementation of the interface?
+	IsTrivial() bool
 }
 
-// A Kernel is a snap.TypeKernel BootParticipant
+// A Kernel exposes functionality that some bootloaders need
 type Kernel interface {
```

Comment:
BootKernel seems a better name for this anyway, then indeed you can have Participant and Kernel

Missing period at the end of the doc comment.

// A BootKernel handles the bootloader setup of a kernel.